miner/worker.go

```diff
@@ -1166,7 +1166,7 @@ func (w *worker) generateWork(params *generateParams, witness bool) *newPayloadR
 	}
 	body := types.Body{Transactions: work.txs, Withdrawals: params.withdrawals}
 	fees := work.state.GetBalance(consensus.SystemAddress)
-	block, receipts, err := w.engine.FinalizeAndAssemble(w.chain, work.header, work.state, &body, work.receipts)
+	block, receipts, err := w.engine.FinalizeAndAssemble(w.chain, work.header, work.state, &body, work.receipts, nil)
```

Comment:
Why not add a tracer here? Same question on line 1432.

Reply:
Miner should not trace block as their are being produced so they cannot be relied on. Their could be a block produced by the miner and a block being processed in state processor at the same time. 

The Geth 1.14.x behaves the same for live tracer(s), they are not invoked by `miner` or optimistic processing.